core/src/main/java/tc/oc/pgm/settings/SettingsMatchModule.java

```diff
@@ -0,0 +1,173 @@
+package tc.oc.pgm.settings;
+
+import com.google.common.collect.Lists;
+import net.kyori.text.Component;
+import net.kyori.text.TranslatableComponent;
+import net.kyori.text.format.TextColor;
+import net.kyori.text.format.TextDecoration;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.ItemFlag;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import tc.oc.pgm.api.PGM;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.match.MatchScope;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.setting.SettingKey;
+import tc.oc.pgm.api.setting.SettingValue;
+import tc.oc.pgm.events.ListenerScope;
+import tc.oc.pgm.settings.tools.FlySpeedTool;
+import tc.oc.pgm.settings.tools.GamemodeTool;
+import tc.oc.pgm.settings.tools.NightVisionTool;
+import tc.oc.pgm.spawns.events.ObserverKitApplyEvent;
+import tc.oc.pgm.util.inventory.ItemBuilder;
+import tc.oc.pgm.util.menu.InventoryMenu;
+import tc.oc.pgm.util.menu.InventoryMenuBuilder;
+import tc.oc.pgm.util.menu.InventoryMenuManager;
+import tc.oc.pgm.util.menu.items.InventoryItem;
+import tc.oc.pgm.util.menu.items.InventoryItemBuilder;
+import tc.oc.pgm.util.text.TextTranslations;
+
+@ListenerScope(MatchScope.LOADED)
+public class SettingsMatchModule implements MatchModule, Listener {
+
+  // Slot where tool item is placed
+  public static final int TOOL_BUTTON_SLOT = 8;
+
+  // Material of tool item item
+  public static final Material TOOL_MATERIAL = Material.DIAMOND;
+
+  public static final String INVENTORY_TITLE = "setting.title";
+  public static final int SETTINGS_ROWS = (SettingKey.values().length + 9 - 1) / 9;
+
+  private final Match match;
+  private InventoryMenuManager manager;
+  private InventoryMenu observerMenu;
+  private InventoryMenu otherMenu;
```

Comment:
Why 2 menus?

Reply:
In-game settings menu doesn't include changing gamemode, flight speed, obs visibility, or night vision, while the observer menu will have all the settings, at least i assume that's the reason

Reply:
^ that's the reason. The inventory item objects themselves are reused across both menus so the memory impact is minimal.